src/utilities/oas-validator.ts

```diff
@@ -137,67 +146,74 @@ class OasValidator {
       if (expectedType === 'array') {
         // check that the actual object is an array
         if (!Array.isArray(actual)) {
-          throw new TypeMismatchError(path, expectedType, actualType);
+          return [new TypeMismatch(path, expectedType, actualType)];
```

Comment:
Since we're returning this now it might make sense to move this check above the enum checks